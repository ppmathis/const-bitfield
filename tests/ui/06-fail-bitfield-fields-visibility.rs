#![feature(const_fn_trait_bound)]
#![feature(const_mut_refs)]
#![feature(const_trait_impl)]

mod inner {
    use const_bitfield::bitfield;

    bitfield! {
        pub struct Test(u8);
        u8, test, set_test: 3, 0;
    }
}

use inner::Test;

pub fn main() {
    let mut test = Test(0);
    test.set_test(0xF);
}
