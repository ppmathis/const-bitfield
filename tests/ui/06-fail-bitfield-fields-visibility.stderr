error[E0624]: associated function `set_test` is private
  --> tests/ui/06-fail-bitfield-fields-visibility.rs:18:10
   |
8  | /     bitfield! {
9  | |         pub struct Test(u8);
10 | |         u8, test, set_test: 3, 0;
11 | |     }
   | |_____- private associated function defined here
...
18 |       test.set_test(0xF);
   |            ^^^^^^^^ private associated function
